Display reports

kryswisnaskas opened the pull request:
**Description of change**
Reports are displayed under "Activity Reports".  In this PR the table contains all reports in smart hub. The UI contains a "New report" button that allows to create a new report. If there are no reports, an empty row will be displayed in the table.

**How to test**
Local:
- pull changes
- yarn deps
- `yarn db:migrate`
- `yarn db:seed`
- `yarn start:local`
- click on "Activity Reports"
- verify the landing page has a table with "Activity reports"
- create a couple of reports
- verify they are displayed

Docker:
- same as above, but with the docker commands

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/5

**Checklist**
<!-- Add details to each completed item -->
- [x] Meets issue criteria
- [x] Code tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [x] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

Note - "Fix Cucumber tests" branch was merged in to potentially avoid cucumber test failures.

jasalisbury commented on frontend/src/pages/Landing/index.js:
```diff
@@ -0,0 +1,331 @@
+import '@trussworks/react-uswds/lib/index.css';
+import './index.css';
+
+export const activityReportId = (id, regionId) => `R${
```
I think this will eventually come from the server once https://github.com/HHS/Head-Start-TTADP/issues/145 is completed

kryswisnaskas replied:
We could implement it on the server if needed.

jasalisbury commented on frontend/src/pages/Landing/index.js:
```diff
@@ -0,0 +1,331 @@
+  return keys.reduce((object, key) => {
+    const result = (object || {})[key];
+    if (key === 'activityRecipients' || key === 'collaborators') {
+      return result[0].name;
```
When sorting by collaborators I get the following error here `TypeError: Cannot read property 'name' of undefined`. I had reports that did and did not have collaborators

kryswisnaskas replied:
Good catch. I refactored name, but didn't update it here.

jasalisbury replied:
:+1: Don't see the error with the newest changes

jasalisbury commented on frontend/src/pages/Landing/index.js:
```diff
@@ -0,0 +1,331 @@
+  });
+
+  const sortedReports = [...reports];
+  const requestSort = (key) => {
```
We'll probably need to move this sort to the API once we start paginating on the server

kryswisnaskas commented on src/routes/activityReports/handlers.js:
```diff
@@ -116,6 +116,21 @@ export async function getReport(req, res) {
+ * @param {*} res - response
+ */
+export async function getReports(req, res) {
+  const reports = await activityReports();
```
We have a separate story to implement pagination (#273). I'll add one for filtering also. Right now we don't filter yet.

jasalisbury replied:
Oh sorry I didn't see that ticket!

jasalisbury reviewed (approved):
Looks good!

dcmcand reviewed (approved):
Nice job Krys!

rahearn commented:
I'll do an in-depth review of this tomorrow morning, but I wanted to flag that when I run `yarn docker:deps` it's changing `yarn.lock` (the backend one). Please update that to be in-sync with `package.json`

rahearn commented on frontend/src/pages/Landing/index.js:
```diff
@@ -0,0 +1,210 @@
+
+    return (
+      <tr key={`landing_${id}`}>
+        <td>
```
Please switch this column to be a `<th scope="row">`

rahearn commented on src/models/user.js:
```diff
@@ -55,6 +55,12 @@ export default (sequelize, DataTypes) => {
+    fullName: {
+      type: DataTypes.VIRTUAL,
+      get() {
+        return `${this.name}, ${this.role.split(' ').map((word) => word[0]).join('')}`;
```
This is causing a 500 error when updating users with a null role. Please deal with a missing role cleanly.

rahearn reviewed (changes requested):
Nice stuff, Krys.

There's two minor accessibility concerns I have with the table, but the biggest issue is that I can't actually see the whole table on my laptop:

<img width="1551" alt="table_cutoff" src="https://user-images.githubusercontent.com/285842/107665611-9c989080-6c5b-11eb-9b36-26ca32e7939e.png">

kryswisnaskas commented on frontend/src/components/LandingLayout.js:
```diff
@@ -0,0 +1,17 @@
+
+function LandingLayout({ children }) {
+  return (
+    <Grid row>
```
The idea was to get away from the `GridContainer` due to layout constraints.

kryswisnaskas commented on frontend/src/pages/Landing/index.js:
```diff
@@ -0,0 +1,210 @@
+          <Alert type="error" role="alert">
+            {error}
+          </Alert>
+        )}
```
Moved.

kryswisnaskas commented:
> What's the purpose of this test vs. the tests that validate each column individually? This feels like maybe it's over-specified

Removed this test.

> Please switch this column to be a <th scope="row">

Switched.

> This is causing a 500 error when updating users with a null role. 

Good catch!
Modified.

> when I run yarn docker:deps it's changing yarn.lock (the backend one). Please update that to be in-sync with package.json

Updated.

kryswisnaskas commented:
>  the biggest issue is that I can't actually see the whole table on my laptop:

I was going to handle that in a separate PR, but since it's already an issue, I added scrollbars now. This wasn't an attempt to hide the right side of the table.

rahearn reviewed (approved):
Code looks good, but the tests failures are consistent for whatever reason. Once that's cleaned up this'll be good to merge
